### PR TITLE
Test/aws backend acls versioning

### PR DESCRIPTION
**Pull request by electrachong**
Just adding some tests to make sure putting/getting acls to objects on AWS backend works properly when versioning is enabled/suspended.

We don't actually modify ACLs on the data in AWS, we rely on our own metadata for ACLs.

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/multipleBackend/get/getAwsVersioning.js [hunk 1]
```diff
@@ -217,12 +217,8 @@ function testSuite() {
                     versionIds.push(...ids);
                     next();
                 },
-                next => getAndAssertResult(s3, {
-                    bucket, key, body: data[4],
-                    versionId: versionIds[4],
-                    expectedVersionId: versionIds[4],
-                },
-                    next),
+                next => getAndAssertResult(s3, { bucket, key, body: data[4],
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/get/getAwsVersioning.js]: why the change?

File: tests/functional/aws-node-sdk/test/multipleBackend/acl/aclAwsVersioning.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    Metadata: { 'scal-location-constraint': awsLocation } },
+    (err, putData) => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `putting object, got error ${err}`);
+        putObjectAcl(s3, key, putData.VersionId, acp, () =>
+            cb(null, putData.VersionId));
+    });
+}
+
+function putVersionsWithAclToAws(s3, key, data, acps, cb) {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/acl/aclAwsVersioning.js]: please do jsdoc for this one. unclear that data is an array of objects and acps are array of different access control policies.

File: tests/functional/aws-node-sdk/test/multipleBackend/acl/aclAwsVersioning.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    this.timeout(30000);
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+
+        beforeEach(() => {
+            process.stdout.write('Creating bucket');
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/acl/aclAwsVersioning.js]: I'd rather if we were testing the location constraint based on the bucket location.

File: tests/functional/aws-node-sdk/test/multipleBackend/get/getAwsVersioning.js (hunk 1 above)
Comment (electrachong) [reply tests/functional/aws-node-sdk/test/multipleBackend/get/getAwsVersioning.js]: the test is supposed to be for getting the correct version without the version id, but actually tested getting with version id. I think no one caught this small detail when reviewing the original PR.

**Comment by electrachong** [comment]
@ironman-machine try

**Comment by electrachong** [comment]
end-to-end auto-retried here; http://ci.ironmann.io/gh/scality/Integration/16198#build-parameters

**Comment by electrachong** [comment]
end to end passed here: http://ci.ironmann.io/gh/scality/Integration/16191#build-parameters
